ambari-logsearch/ambari-logsearch-server/src/main/java/org/apache/ambari/logsearch/common/PropertiesSplitter.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.ambari.logsearch.common;
+
+import com.google.common.base.Splitter;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.StringUtils;
+
+import javax.inject.Named;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Named
+public class PropertiesSplitter {
+
+  public List<String> parseList(String listStr) {
+    return parseList(listStr, ",");
+  }
+
+  public Map<String, String> parseMap(String mapStr) {
+    return parseMap(mapStr, ",", ":");
+  }
+
+  private List<String> parseList(String listStr, String separator) {
+    return StringUtils.isNotBlank(listStr) ? Splitter.on(separator).splitToList(listStr) : new ArrayList<>();
+  }
+
+  public Map<String, String> parseMap(String mapStr, String separator, String keyValueSeparator) {
```

Comment:
This may be can done with Splitters as well:
Splitter.on(separator)
              .trimResults()
              .withKeyValueSeparator(Splitter.on(keyValueSeparator).limit(2).trimResults())
              .split(mapStr)

Reply:
i kept my solution as Maps with Splitter create key value pairs with null key as well